balance: Полностью автоматический C-20rm для нюкеров

vovakons opened the pull request:
## Что этот ПР делает

Добавляем полностью автоматический вариант C-20r SMG. Доступен только в аплинке нюкеров.

Новый смг называется C-20rm (*modified*) улучшенная версия с режимом автоогня. Задержка между выстрелами 0.15 секунд (*у ВТ 0.2 секунды*).

## Почему это хорошо для игры

[Обсуждение в дискорде](https://discord.com/channels/617003227182792704/1406597999953580032/1406597999953580032)

## Тестирование
Локалка, в аплинке теперь автоматический вариант, автоогонь быстрее чем у вт.

vovakons committed:
add full auto C-20rm

vovakons committed:
c-20rm for nuclear uplink

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -132,6 +132,20 @@
 	icon_state = "c20r[magazine ? "-[CEILING(get_ammo(FALSE)/4, 1)*4]" : ""][chambered ? "" : "-e"][suppressed ? "-suppressed" : ""]"
 
 
+//C-20rm Full auto//
```
```suggestion
// C-20rm Full auto
```

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -132,6 +132,20 @@
+//C-20rm Full auto//
+/obj/item/gun/projectile/automatic/c20r/auto
+	name = "C-20rm Full-Auto SMG"
+	desc = "A full-auto .45 SMG, modification of 'C-20r'. Has a 'Scarborough Arms - Per falcis, per pravitas' buttstamp."
```
??

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -132,6 +132,20 @@
 
+//C-20rm Full auto//
+/obj/item/gun/projectile/automatic/c20r/auto
+	name = "C-20rm Full-Auto SMG"
```
Рунеймсы

vovakons committed:
localization

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -132,6 +132,30 @@
+//C-20rm Full auto
+/obj/item/gun/projectile/automatic/c20r/auto
+	name = "C-20rm Full-Auto SMG"
+	desc = "Автоматический пистолет-пулемет под .45 калибр. Новейшая модификация пистолет-пулемета 'C-20r' ."
```
```suggestion
	desc = "Новейшая модификация автоматического пистолет-пулемета \"C-20r\" под .45 калибр."
```

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -132,6 +132,30 @@
 
+//C-20rm Full auto
+/obj/item/gun/projectile/automatic/c20r/auto
+	name = "C-20rm Full-Auto SMG"
```
фулл авто точно нужно в названии? другого C-20rm нет

vovakons committed:
localization #2

vovakons committed:
Merge branch 'master220' into c-20r-auto

vovakons committed:
c-20rm autofire with firemodes

Удален старый способ давать автоогонь ппшке C-20rm.
Добавлен автоогонь для C-20rm через новые режимы стрельбы у автоматических пушек.

dageavtobusnick committed:
Merge branch 'master220' into c-20r-auto

littleboobs committed:
Merge branch 'master220' into c-20r-auto

vovakons commented:
@dageavtobusnick было бы неплохо и это замержить, сейчас цр сильно уступает вт

ROdenFL commented on code/datums/uplink_item.dm:
```diff
@@ -803,9 +803,9 @@
 	desc = "Полностью заряженный пистолет-пулемёт, оснащённый магазином на 20 патронов .45 калибра. \
-			Имеет два режима стрельбы: полуавтоматический и с отсечкой по 2 патрона. Совместим с глушителем."
+			Имеет только автоматический режим огня. Совместим с глушителем."
 	item = /obj/item/gun/projectile/automatic/c20r
```
Мне кажется или ты путь обновить забыл

vovakons replied:
есть такое

dageavtobusnick commented:
> @dageavtobusnick было бы неплохо и это замержить, сейчас цр сильно уступает вт

Ужас... Сделай обоим цркам автоогонь, просто у улучшеной сделай скорострельность выше.

ROdenFL commented:
> > @dageavtobusnick было бы неплохо и это замержить, сейчас цр сильно уступает вт
> 
> Ужас... Сделай обоим цркам автоогонь, просто у улучшеной сделай скорострельность выше.

Так он ничего никуда и не сделал...
Он создал новый ган и никуда его не добавил...

dageavtobusnick commented:
> > > @dageavtobusnick было бы неплохо и это замержить, сейчас цр сильно уступает вт
> > 
> > 
> > Ужас... Сделай обоим цркам автоогонь, просто у улучшеной сделай скорострельность выше.
> 
> Так он ничего никуда и не сделал... Он создал новый ган и никуда его не добавил...

Хдд. Я код ещё не смотрел ещё

vovakons commented:
> Так он ничего никуда

ну мапперы у нас накидали црок в лут, не очень хочется потом видеть гейтеров или механиков с цркой на автоогне

vovakons committed:
fix item for nuke

vovakons committed:
Merge branch 'master220' into c-20r-auto

littleboobs committed:
Merge branch 'master220' into c-20r-auto

littleboobs committed:
Merge branch 'master220' into c-20r-auto

dageavtobusnick commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -162,6 +162,26 @@
+	desc = "Новейшая модификация автоматического пистолет-пулемета \"C-20r\" под .45 калибр."
+	accuracy = GUN_ACCURACY_PISTOL
+	recoil = GUN_RECOIL_LOW
+	fire_modes = GUN_MODE_SINGLE_BURST_AUTO
```
Сделай у обоих ЦРок ауто, просто у базовой меньше скорострельность в разы. Возможно потом ее закинем в триторский аплинк, взамен говнопистолетам.

vovakons committed:
Merge branch 'master220' into c-20r-auto

vovakons committed:
Merge branch 'c-20r-auto' of github.com:vovakons/Paradise into c-20r-auto

vovakons committed:
c20-r auto mode

Добавлен автоматический режим стрельбы в обычный C-20r, но с более
низким темпом стрельбы.

vovakons committed:
remove duplicate variable
